adaptdl/adaptdl/torch/scaling_rules.py

```diff
@@ -125,6 +125,41 @@ def scale_lr(self, scale):
         return (var + sqr) / (var / scale + sqr)
 
 
+class AdamScale(ScalingRuleBase):
+    """
+    Implements the variant of AdaScale_ that supports Adam, AdamW and RMSProp
+    """
+
+    def scale_lr(self, scale,  power=0.5):
+        """Calculate factors to be applied to lr for each parameter group."""
+        var = self.adp.gns.raw_var_avg
+        sqr = self.adp.gns.raw_sqr_avg
+        var = np.maximum(var, 1e-6)
+        sqr = np.maximum(sqr,  0.0)
+        return np.power((var + sqr) / (var / scale + sqr), power)
+
+    def step(self, *args, **kwargs):
+        if not self.adp:
+            raise ValueError("AdaptiveDataParallel instance is not set!")
+        if not self.adp.require_backward_grad_sync:
+            return
+        scale = self.adp.gns.accum_scale * self.adp.gns.accum_count
+        if self.adp.gns._is_adam and \
```

Comment:
Is there ever a case when the AdamScale class is used and the gns doesnt have the adam flag set?